vllm/model_executor/parallel_utils/tensor_parallel/layers.py

```diff
@@ -376,32 +407,55 @@ def __init__(self, input_size, output_size, *,
         self.world_size = get_tensor_model_parallel_world_size()
         self.input_size_per_partition = divide(input_size, self.world_size)
         self.skip_bias_add = skip_bias_add
+        self.quant_config = quant_config
+
+        if self.quant_config is None:
+            # Parameters.
+            # Note: torch.nn.functional.linear performs XA^T + b and as a result
+            # we allocate the transpose.
+            # Initialize weight.
+            if use_cpu_initialization:
+                self.weight = Parameter(torch.empty(self.output_size,
+                                                    self.input_size_per_partition,
+                                                    dtype=params_dtype))
+                if perform_initialization:
+                    self.master_weight = _initialize_affine_weight_cpu(
+                        self.weight, self.output_size, self.input_size,
+                        self.input_size_per_partition, 1, init_method,
+                        stride=stride, return_master_weight=keep_master_weight_for_test)
+            else:
+                self.weight = Parameter(torch.empty(
+                    self.output_size, self.input_size_per_partition,
+                    device=torch.cuda.current_device(), dtype=params_dtype))
+                if perform_initialization:
+                    _initialize_affine_weight_gpu(self.weight, init_method,
+                                                  partition_dim=1, stride=stride)
+            self.register_parameter('qweight', None)
+            self.register_parameter('qzeros', None)
+            self.register_parameter('scales', None)
+        else:
+            # Quantized parameters.
+            # TODO(julian-q) add initialization?
```

Comment:
As we discussed we don't need initialization here and we'll refactor the code to remove the initialization above as well.